### ISSUE TITLE
should UA style sheet have a rule for visited link color?

### ISSUE THREAD

**Issue by heycam** [w3c/svgwg#47]
@espadrine [writes](https://lists.w3.org/Archives/Public/www-svg/2016Feb/0025.html):

> The SVG 1.1 and 2 specs mention that implementations must respect CSS2 (resp. 2.1) pseudo-classes including :visited (https://www.w3.org/TR/SVG2/styling.html#RequiredCSSFeatures).
> 
> The default UA style sheet does not specify a :visited rule (https://www.w3.org/TR/SVG/styling.html#DefaultStyleSheetLanguage). There doesn't seem to be an explicit requirement not to have a :visited rule, even though the absence of a requirement could be interpreted as a hint not to do so.
> 
> Regardless, Blink has a default :visited rule (or the behavior of one), causing incompatibilities (https://code.google.com/p/chromium/issues/detail?id=352912, see also https://jsbin.com/nixexe/ for example). In Gecko and WebKit, `<g fill="#00ff00"><a xlink:href="http://google.com">Text</a></g>` will show as green. In Blink, it will only be green if the link has not been visited, black otherwise, leading to subtle bugs.
> 
> As stated on the bug report, the Blink authors justify their implementation with the spec's ambiguity:
> 
> > Yes, it could be argued that the UA style for <svg:a> should be the same as <html:a> (although that's not mandated by the spec AFAIK.)
> 
> In order to erase that ambiguity and prevent future incompatibilities, we should specify that default UA style rules not present in 6.16 should not be enforced by default by UAs.
> 
> What are your opinions?

**Comment by heycam** [comment]
It's not clear to me why the text of the visited link becomes black in Chrome.  Is there a `svg|*:visited { fill: black; }` rule in their UA style sheet?

I would be in favour of adding a note like:

> Note that the user agent style sheet above does not contain any rules to style elements that match the :link or :visited pseudo-classes. To ensure SVG documents with links are rendered interoperably, user agents should not have rules that affect the styling of elements in the SVG namespace that match these pseudo-classes.

**Comment by dkturner** [comment]
It appears that Chrome does indeed have such a rule in its UA style sheet, because adding `svg a:visited { fill: inherit; }` in a user style sheet solves the problem.

**Comment by heycam** [comment]
@progers what do you think about removing the UA style sheet rule?

**Comment by progers** [comment]
That seems very reasonable to me, we were just looking for spec clarification.

I poked around and couldn't find this in our user agent stylesheets so I think this is actually just a flat-out bug. I'll followup on the chrome bug.

(Btw, I am liking using an issue tracker for spec bugs. I've just clicked a button to get updates, no need for special accounts, per-thread views are handled much better, etc. I suspect we'll be more productive since discussion will stay on topic more in an issue tracker interface.)
